internal/testclient/testtx/context.go

```diff
@@ -0,0 +1,350 @@
+package testtx
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"cosmossdk.io/depinject"
+	abci "github.com/cometbft/cometbft/abci/types"
+	cometbytes "github.com/cometbft/cometbft/libs/bytes"
+	cometrpctypes "github.com/cometbft/cometbft/rpc/core/types"
+	comettypes "github.com/cometbft/cometbft/types"
+	cosmosclient "github.com/cosmos/cosmos-sdk/client"
+	cosmostx "github.com/cosmos/cosmos-sdk/client/tx"
+	cosmoskeyring "github.com/cosmos/cosmos-sdk/crypto/keyring"
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+
+	"pocket/internal/mocks/mockclient"
+	"pocket/internal/testclient"
+	"pocket/pkg/client"
+	"pocket/pkg/client/tx"
+)
+
+// NewOneTimeErrTxTimeoutTxContext creates a mock transaction context designed to simulate a specific
+// timeout error scenario during transaction broadcasting.
+//
+// Parameters:
+// - t: The testing.T instance for the current test.
+// - keyring: The Cosmos SDK keyring containing the signer's cryptographic keys.
+// - signingKeyName: The name of the key within the keyring to use for signing.
+// - expectedTx: A pointer whose value will be set to the expected transaction
+// bytes (in hexadecimal format).
+// - expectedTxHash: A pointer whose value will be set to the expected
+// transaction hash.
+// - expectedErrMsg: A pointer whose value will be set to the expected error
+// message string.
+//
+// The function performs the following actions:
+// 1. It retrieves the signer's cryptographic key from the provided keyring using the signingKeyName.
+// 2. It computes the corresponding address of the signer's key.
+// 3. It then formats an error message indicating that the fee payer's address does not exist.
+// 4. It creates a base mock transaction context using NewBaseTxContext.
+// 5. It sets up the mock behavior for the BroadcastTxSync method to return a specific preset response.
+// 6. It also sets up the mock behavior for the QueryTx method to return a specific error response.
+func NewOneTimeErrTxTimeoutTxContext(
+	t *testing.T,
+	keyring cosmoskeyring.Keyring,
+	signingKeyName string,
+	expectedTx *cometbytes.HexBytes,
+	expectedTxHash *cometbytes.HexBytes,
+	expectedErrMsg *string,
+) *mockclient.MockTxContext {
+	t.Helper()
+
+	signerKey, err := keyring.Key(signingKeyName)
+	require.NoError(t, err)
+
+	signerAddr, err := signerKey.GetAddress()
+	require.NoError(t, err)
+
+	*expectedErrMsg = fmt.Sprintf(
+		"fee payer address: %s does not exist: unknown address",
+		signerAddr.String(),
+	)
+
+	txCtxMock := NewBaseTxContext(
+		t, signingKeyName,
+		keyring,
+		expectedTx,
+		expectedTxHash,
```

Comment:
Not actionable, just sharing an idea for the future.

Rather than providing `expectedTxHash` and `expectedTx` as separate params, we could provide a function that transforms `expectedTx` to `expectedTxHas`. Probably overkill for now but just planting the seed.

Reply:
:ok_hand: